### PR TITLE
Assignment 2

### PR DESCRIPTION
**Pull request by mhall903**
Please review my code to merge with master.

### CODE REVIEW

File: PressureSensorAdapter.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        this.pressure = (double)value;
+        return this.pressure;
+    }
+    @Override
+    public String getReport(){
+        String report;
+        if(this.pressure>6.58){
+            report="DANGER -->"+this.pressure;
+            return report;
+        }
```
Comment (BrandaeGrein) [inline PressureSensorAdapter.java]: this is great

File: PressureSensorAdapter.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import java.util.Random;
```
Comment (nganngo) [inline PressureSensorAdapter.java:1]: The adapter classes are redundant from rewrite each methods from Sensor class. Improve example:
@Override
    public String getName() {
            return adaptee.getName();
}

File: SensorApplication.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            progressBar.setForeground(Color.yellow);
+        }
+        Border border = BorderFactory.createTitledBorder(name);
+        progressBar.setBorder(border);
+        content.add(progressBar, BorderLayout.NORTH);
+        content.add(tdisplay, BorderLayout.CENTER);
+        f.setSize(600, 600);
+        f.setVisible(true);
+    }
+    public static void main(String[] args) {
```
Comment (nganngo) [inline SensorApplication.java:55]: Create an array of Sensor to reduce repeating lines of code.

File: SensorApplication.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
+
```
Comment (nganngo) [inline SensorApplication.java:1]: Formatting is very nice to follow and understand the code.

**Review by nganngo — APPROVED** [review]

File: PressureSensorAdapter.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import sensor.PressureSensor;
+public class PressureSensorAdapter implements SensorInterface{
+    private PressureSensor adaptee;
+    private double pressure;
+
+    public PressureSensorAdapter(){
+
+    }
+    @Override
+    public double readValue(){
```
Comment (BrandaeGrein) [inline PressureSensorAdapter.java:11]: This is the adapter class. Adapter classes shouldn't have any implementation in them. 

Here's an example:
public double readValue(){
adaptee.readValue();
}
public String getReport(){
adaptee.getReport();
}

and so on.

File: RadiationSensorAdapter.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
+import java.util.Random;
```
Comment (BrandaeGrein) [inline RadiationSensorAdapter.java:1]: I think we have different definitions of adapter classes. Here's the video that I followed -- it gives a good explanation:
https://www.youtube.com/watch?v=qG286LQM6BU&t=530s

File: SensorApplication.java [hunk 7]
```diff
@@ -0,0 +1,81 @@
+
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.List;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.border.TitledBorder;
+
```
Comment (BrandaeGrein) [inline SensorApplication.java:9]: Unnecessary space

File: SensorApplication.java [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.border.TitledBorder;
+
+import javax.swing.BorderFactory;
+import javax.swing.JFrame;
+import javax.swing.JProgressBar;
+import javax.swing.border.Border;
+
+public class SensorApplication extends JFrame {
```
Comment (BrandaeGrein) [inline SensorApplication.java:15]: Please add comments for readability.

File: SensorApplication.java [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+            progressBar.setMaximum(400);
+
+        }
+        else if(name.contains("Pressure")){
+            progressBar.setMaximum(10);
+
+        }
+        else {
+            progressBar.setMaximum(5);
+
```
Comment (BrandaeGrein) [inline SensorApplication.java:34]: Unnecessary Space

File: SensorApplication.java [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+        TextArea tdisplay=new TextArea(1,15);
+        tdisplay.append(report);
+
+        if(name.contains("Temperature")){
+            progressBar.setMaximum(400);
+
+        }
+        else if(name.contains("Pressure")){
+            progressBar.setMaximum(10);
+
```
Comment (BrandaeGrein) [inline SensorApplication.java:30]: Unnecessary Space

File: SensorApplication.java [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+        JFrame f = new JFrame("Sensor Tracker");
+        f.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        Container content = f.getContentPane();
+        JProgressBar progressBar = new JProgressBar();
+        TextArea tdisplay=new TextArea(1,15);
+        tdisplay.append(report);
+
+        if(name.contains("Temperature")){
+            progressBar.setMaximum(400);
+
```
Comment (BrandaeGrein) [inline SensorApplication.java:26]: Unnecessary Space

File: TemperatureSensorAdapter.java [hunk 12]
```diff
@@ -0,0 +1,40 @@
+import sensor.TemperatureSensor;
+import java.util.Random;
+
+public class TemperatureSensorAdapter implements SensorInterface {
```
Comment (BrandaeGrein) [inline TemperatureSensorAdapter.java:4]: Again, I don't think this is quite following the adapter class formula.

**Review by BrandaeGrein — COMMENTED** [review]
Overall, it's good code. It's got solid object oriented design -- right now I think it's not quite the adapter class layout, but that's easily fixed. The video I provided should help.

**Pull request merged**